util/network-devp2p/src/discovery.rs

```diff
@@ -457,61 +471,105 @@ impl<'a> Discovery<'a> {
 		entry.endpoint.is_allowed(&self.ip_filter) && entry.id != self.id
 	}
 
-	fn on_ping(&mut self, rlp: &Rlp, node: &NodeId, from: &SocketAddr, echo_hash: &[u8]) -> Result<Option<TableUpdates>, Error> {
+	fn on_ping(&mut self, rlp: &Rlp, node_id: &NodeId, from: &SocketAddr, echo_hash: &[u8]) -> Result<Option<TableUpdates>, Error> {
 		trace!(target: "discovery", "Got Ping from {:?}", &from);
-		let source = NodeEndpoint::from_rlp(&rlp.at(1)?)?;
-		let dest = NodeEndpoint::from_rlp(&rlp.at(2)?)?;
+		let ping_from = NodeEndpoint::from_rlp(&rlp.at(1)?)?;
+		let ping_to = NodeEndpoint::from_rlp(&rlp.at(2)?)?;
 		let timestamp: u64 = rlp.val_at(3)?;
 		self.check_timestamp(timestamp)?;
 
 		let mut response = RlpStream::new_list(3);
-		dest.to_rlp_list(&mut response);
+		let pong_to = NodeEndpoint {
```

Comment:
Why is this different from `ping_from`? It makes more sense to me to respond to the endpoint that the sender claims is their address than mixing and matching it with the `from` SocketAddr.

Reply:
Actually, this isn't really used from the implementations that I've read, but I think the idea is that the recipient of the Pong knows what we think his address is. There is no point in copying a field of the packet he just sent us, he probably knows it already.

Reply:
Well, it's used below as the entry that `add_node` is called on. Whereas I think `ping_from` is a better choice.

Reply:
Sorry I thought it was to `on_pong` method.

Reply:
It would be actually quite a bad idea to use the `ping_from` that we have been sent, since it's mostly the peer local address, such as `V4(0.0.0.0:42786)` or `V6([::]:30303)`